solo/admin.py

```diff
@@ -1,25 +1,30 @@
-from django.urls import re_path
+from __future__ import annotations
+
+from typing import Any
+
+from django.db.models import Model
+from django.urls import URLPattern, re_path
 from django.contrib import admin
-from django.http import HttpResponseRedirect
+from django.http import HttpRequest, HttpResponse, HttpResponseRedirect
 from django.utils.encoding import force_str
 from django.utils.translation import gettext as _
 
 from solo.models import DEFAULT_SINGLETON_INSTANCE_ID
 from solo import settings as solo_settings
 
 
-class SingletonModelAdmin(admin.ModelAdmin):
+class SingletonModelAdmin(admin.ModelAdmin):  # type: ignore[type-arg]
```

Comment:
Would using [`django-stubs`](https://github.com/typeddjango/django-stubs) help avoid any of these `# type: ignore`'s?

We'd only want `django-stubs` to be a dev dependency though, not a full one. Perhaps only installed via tox (probably need to pin it in tox config or using something like [`pip-tools`](https://github.com/jazzband/pip-tools).

Reply:
Oh, I did see you added `django-stubs` later in the PR. Any idea why these and other Django related things seem to still need `# type: ignore`? If there is something missing from `django-stubs`, could potentially open an issue over there. That team is quite responsive.

Reply:
So basically `django-stubs` is making use of generics when possible, and `ModelAdmin` is an example. The main drawback is it is only defined as being generic in the stubs, not at runtime. A placeholder `__class_getitem__` was added upstream for some classes (I think `Manager` and some fields), but `ModelAdmin` is still lacking.

The workaround for application developers is to use [`django-stubs-ext`](https://github.com/typeddjango/django-stubs/tree/master/ext) that will monkeypatch Django at runtime and add `__class_getitem__` definitions when required. However, this does not apply to library development.

I think it could be nice to go with the following:

```python
from django.contrib.admin import ModelAdmin as _ModelAdmin

if TYPE_CHECKING:
    ModelAdmin = _ModelAdmin
else:
    class ModelAdmin(_ModelAdmin):
        def __class_getitem__(cls, *args, **kwargs):
            return cls

_ModelT = TypeVar("_ModelT", bound=SingletonModel)

class SingletonModelAdmin(ModelAdmin[_ModelT]):
    ...
```

But this is quite verbose.

Reply:
With the current solution, users will not be able to do the following:

```python
class MySingletonAdmin(SingletonModelAdmin[MySingleton]):
    ...
```

However, this is a niche feature, I don't see a lot of people making use of it (see this [search result](https://github.com/search?q=language%3Apython+%22ModelAdmin%5B%22&type=code)). Let's keep it as it is for now and if people complain about it the workaround above could still be added